### PR TITLE
Set-functions: rework to raise instead of return

### PR DESCRIPTION
**Pull request by bouwew**
Also:
- move error-messages into the raises
- cleanup unneeded try-excepts, already handled in _request()

### CODE REVIEW

File: plugwise/smile.py [hunk 1]
```diff
@@ -320,38 +317,31 @@ async def _smile_detect_legacy(self, result: etree, dsmrmain: etree) -> str:
         if result.find('./appliance[type="thermostat"]') is None:
             # It's a P1 legacy:
             if dsmrmain is not None:
-                try:
-                    status = await self._request(STATUS)
-                    self.smile_fw_version = status.find("./system/version").text
-                    model = status.find("./system/product").text
-                    self.smile_hostname = status.find("./network/hostname").text
-                    self.smile_mac_address = status.find("./network/mac_address").text
-                except InvalidXMLError:  # pragma: no cover
```
Comment (CoMPaTech) [inline plugwise/smile.py:329]: agreed on removing this, but I recall we really had a p1 corner case, lets see if it reoccurs

File: tests/test_smile.py [hunk 2]
```diff
@@ -463,13 +463,15 @@ async def tinker_switch(
         for new_state in [False, True, False]:
             _LOGGER.info("- Switching %s", new_state)
             try:
-                switch_change = await smile.set_switch_state(
-                    dev_id, members, model, new_state
-                )
+                await smile.set_switch_state(dev_id, members, model, new_state)
+                switch_change = True
+            except pw_exceptions.PlugwiseError:
```
Comment (CoMPaTech) [inline tests/test_smile.py:468]: Better!

**Review by CoMPaTech — APPROVED** [review]
